tsne/bh_sne_src/tsne.cpp

```diff
@@ -576,11 +587,12 @@ double randn() {
   return x;
 }
 
-// Perform t-SNE
+// Initialize t-SNE
 template <int NDIMS>
-void run(double* X, int N, int D, double* Y, double perplexity, double theta,
-         int rand_seed, bool skip_random_init, double* init, bool use_init,
-         int max_iter, int stop_lying_iter, int mom_switch_iter) {
+struct TSNE* init_tsne(double* X, int N, int D, double* Y, double perplexity,
```

Comment:
I'd be a lot more comfortable with this using
```suggestion
unique_ptr<TSNEState> init_tsne(double* X, int N, int D, double* Y, double perplexity,
```
Though I do understand that the C api makes that a bit tricky.  It can always use `.get()`, though, thus isolating the badness to inside the `extern "C"` block.

Reply:
you mean `.release()`, yes?